db/rdbms/src/main/resources/mapper/Commons.xml

```diff
@@ -78,4 +78,88 @@
     </if>
   </sql>
 
+  <sql id="variableOperationCondition">
+    <choose>
+      <when test="operation.operator.name().equals('EQUALS')">
+        <if test="operation.type.toString == 'STRING' or operation.type.toString == 'BOOLEAN'">
+          VAR_VALUE = #{operation.value}
+        </if>
+        <if test="operation.type.toString == 'LONG'">
+          LONG_VALUE = #{operation.value}
+        </if>
+        <if test="operation.type.toString == 'DOUBLE'">
+          DOUBLE_VALUE = #{operation.value}
+        </if>
+      </when>
+      <when test="operation.operator.name().equals('NOT_EQUALS')">
+        <if test="operation.type.toString == 'STRING' or operation.type.toString == 'BOOLEAN'">
+          VAR_VALUE != #{operation.value}
+        </if>
+        <if test="operation.type.toString == 'LONG'">
+          LONG_VALUE != #{operation.value}
+        </if>
+        <if test="operation.type.toString == 'DOUBLE'">
+          DOUBLE_VALUE != #{operation.value}
+        </if>
+      </when>
+      <when test="operation.operator.name().equals('EXISTS')">
+        <!-- can be ignored, doesn't make sence for searching of single variables -->
+        1 = 1
+      </when>
+      <when test="operation.operator.name().equals('NOT_EXISTS')">
+        <!-- can be ignored, doesn't make sence for searching of single variables -->
+        1 = 1
+      </when>
+      <when test="operation.operator.name().equals('GREATER_THAN')">
+        <if test="operation.type.toString == 'LONG'">
+          LONG_VALUE &gt; #{operation.value}
+        </if>
+        <if test="operation.type.toString == 'DOUBLE'">
+          DOUBLE_VALUE &gt; #{operation.value}
+        </if>
```

Comment:
❓ What happens if the GREATER_THAN is performed on a STRING? Is it simply ignored or do we run in an error?

Reply:
Depends on the database. But at least when the UntypedOperation is used, that should not happen.